### DISCUSSION TITLE
Support hiding the trailing / final newline

### DISCUSSION THREAD

**Discussion by ModProg** [helix-editor/helix#8365] [General]
#8157 added `insert-final-newline`, something that I personally like, but I prefer the way vim handles it, by hiding this trailing newline.
 
 I would like to have this trailing newline not show up as an additional empty line. (this could/should be configurable)

**Comment by pascalkuthe** [comment]
This is something we rejected repeatedly and orthogonal to that option. It's an intentional part of helix editing model that the last line is selectable.

**Reply by ModProg** [reply]
I assume this is the same reason you can move past the last character in a line (i.e., on the `\n`).

I didn't find much on the reasoning on this. The only thing I found is to make `t<ret>` work, but I feel like that would be easy to support otherwise as well (e.g., by mapping `t<ret>` to `vgl`).

I understand that this is very much my personal preference coming from vim, and hopefully I'll be able to implement a workaround when the plugin support lands.

(I would be willing to implement this as a builtin config option my self as well, but I also understand you not having to maintain that extra code, for a feature you disagree with.)

**Reply by reinux** [reply]
This really feels like a poor choice on so many levels.
- LSPs have to take into account being fed a position to a line that doesn't exist, which can hamper compatibility
- The user needs to check, at least visually (but by default by prodding) to see if their cursor is on the newline or a whitespace

**Comment by kirawi** [comment]
In addition to what Pascal said, I think this was also a decision solidified in relation to https://github.com/helix-editor/helix/issues/362 though I forget where this was mentioned.

Edit: https://github.com/helix-editor/helix/issues/309

**Reply by ModProg** [reply]
Well for EOF my preference is more of cosmetic nature, (i.e., I don't think it would impact anything if I just couldn't move into the phantom line below the last actual line).

For EOL behavior, this is probably not changeable, because while I personally don't see a reason to move on the new line in normal mode (if you want to remove it, I'd just join the lines, if I want to insert something at the start of the next line, I'll just do `jI`. ), it probably does make sense to be able to “selectable” and helix doesn't make a difference here. So I probably just have to get used to it.
